tests/js/client/communication/test-chaos-cluster.js

```diff
@@ -0,0 +1,290 @@
+            } 
+          }
+        };
+        
+        let query = db._query;
+        let trx = null;
+        let ctx = db;
+        if (testOpts.withStreamingTransactions && Math.random() < 0.5) {
+          trx = db._createTransaction({ collections: { write: [c.name()] } });
+          ctx = trx;
+          c = trx.collection(testOpts.collection);
+          query = trx.query;
+        }
+
+        const ops = Math.round(Math.random() * 5) + 1;
+        for (let op = 0; op < ops; ++op) {
+          try {
+            const d = Math.random();
+            if (d >= 0.98 && testOpts.withTruncate) {
+              c.truncate();
+            } else if (d >= 0.9) {
+              query.call(ctx, "FOR doc IN @docs INSERT doc INTO " + c.name(), {docs: docs()}, opts());
+            } else if (d >= 0.8) {
+              query.call(ctx, "FOR doc IN " + c.name() + " LIMIT 138 REMOVE doc IN " + c.name(), {}, opts());
+            } else if (d >= 0.7) {
+              query.call(ctx, "FOR doc IN " + c.name() + " LIMIT 1338 REPLACE doc WITH { pfihg: 434, fjgjg: 555 } IN " + c.name(), {}, opts());
+            } else if (d >= 0.25) {
+              c.insert(docs(), opts());
+            } else {
+              c.remove(docs());
+            }
+          } catch (err) {}
+        }
+        
+        if (trx) {
+          if (Math.random() < 0.2) {
+            trx.abort();
+          } else {
+            trx.commit();
+          }
+        }
+      };
+
+      testOpts.collection = cn;
+      code = `(${code.toString()})(${JSON.stringify(testOpts)});`;
+      
+      let tests = [];
+      for (let i = 0; i < 3; ++i) {
+        tests.push(["p" + i, code]);
+      }
+
+      // run the suite for 5 minutes
+      runParallelArangoshTests(tests, 5 * 60, cn);
+
+      print("Finished load test - waiting for cluster to get in sync before checking consistency.");
+      clearAllFailurePoints();
+      waitForShardsInSync(cn);
+      checkCollectionConsistency(cn);
+    }
+  };
+}
+
+function BuildChaosSuite(opts, suffix) {
+  let suite = {};
+  deriveTestSuite(BaseChaosSuite(opts), suite, suffix);
+  return suite;
+}
+
+const params = ["IntermediateCommits", "FailurePoints", "Truncate", "VaryingOverwriteMode", "StreamingTransactions"];
+
+function addSuite(paramValues) {
+  let suffix = "";
+  let opts = {};
+  for (let j = 0; j < params.length; ++j) {
+    suffix += paramValues[j] ? "_with_" : "_no_";
+    suffix += params[j];
+    opts["with" + params[j]] = paramValues[j];
+  }  
+  let func = function() { return BuildChaosSuite(opts, suffix); };
+  // define the function name as it shows up as suiteName
+  Object.defineProperty(func, 'name', {value: "ChaosSuite" + suffix, writable: false});
+
+  jsunity.run(func);
+}
+/*
+for (let i = 0; i < (1 << params.length); ++i) {
+  let paramValues = [];
+  for (let j = params.length - 1; j >= 0; --j) {
+    paramValues.push(Boolean(i & (1 << j)));
+  }
+  addSuite(paramValues);
+}
+*/
```

Comment:
@mpoeter wanted to handle this but apparently didn't.
It should still be commented out but will be activated soon, separately. Let's leave it as it is for now.

Reply:
I added a comment in the code two days ago, not sure why it was not in the diff yesterday. 

This code dynamically creates test suites for all parameter combinations. ATM we don't use it since we don't want to include ALL suites in the PR tests, but we do want to have them in the nightlies. I am currently working on a separate PR for that which will replace this test entirely.